ts/features/payments/checkout/utils/index.ts

```diff
@@ -36,3 +42,50 @@ export const getLatestUsedWallet = (
     ),
     O.fromNullable
   );
+
+export const WalletPaymentStepScreenNames = {
+  [WalletPaymentStepEnum.PICK_PAYMENT_METHOD]: "PICK_PAYMENT_METHOD",
+  [WalletPaymentStepEnum.PICK_PSP]: "PICK_PSP",
+  [WalletPaymentStepEnum.CONFIRM_TRANSACTION]: "CONFIRM_TRANSACTION"
+};
+
+export const getPspFlagType = (
+  psp: Bundle,
+  pspList?: ReadonlyArray<Bundle>
+): PaymentAnalyticsSelectedPspFlag => {
+  if (!pspList) {
+    return "none";
+  }
+  if (psp.onUs) {
+    return "customer";
+  }
+  if (pspList.length === 1) {
+    return "unique";
+  }
+  const cheaperPsp = pspList.reduce((acc, curr) => {
+    if (
+      acc.taxPayerFee &&
+      curr.taxPayerFee &&
+      acc.taxPayerFee < curr.taxPayerFee
+    ) {
+      return acc;
+    }
+    return curr;
+  });
```

Comment:
What about using a simple sort by taxPayerFee and selecting the first in the array?

```suggestion
  const cheaperPsp = _.orderBy(pspList, psp => psp.taxPayerFee)[0];
```

Reply:
Addressed into 46485c04bf8b8b62fed3539b69e2640eaed65948